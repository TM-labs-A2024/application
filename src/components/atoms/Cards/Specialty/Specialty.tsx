import { Link } from '@chakra-ui/next-js'
import { Text, Avatar } from '@chakra-ui/react'
import { Patient } from '@src/types'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Specialty({
  name,
  id,
  patient
}: {
  name: string
  id: string
  patient?: Patient
}) {
  return (
    <Link
      as={NextLink}
      href={patient ? `/especialidad/${patient?.id}/${id}` : `/especialidad/${id}`}
      className="border-1 flex w-full rounded-md border border-gray-400 p-4"
      data-testid="specialty-card"
    >
      <div className="flex w-full flex-row items-center justify-between">
        <div className="flex flex-row items-center">
          <Avatar name={name} src="/" />
          <Text className="ml-4">{name}</Text>
        </div>
        <Chevron />
      </div>
    </Link>
  )
}
