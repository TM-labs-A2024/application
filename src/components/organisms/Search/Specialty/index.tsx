import SpecialtySearch from './SpecialtySearch'

export default SpecialtySearch
