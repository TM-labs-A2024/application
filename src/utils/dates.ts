export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
