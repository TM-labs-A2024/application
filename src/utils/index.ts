import { sendEmail } from './email'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export { isWeb, isIOS, isAndroid, isMobile, sendEmail }
