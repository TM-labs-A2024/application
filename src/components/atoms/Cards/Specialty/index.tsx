import Specialty from './Specialty'

export default Specialty
