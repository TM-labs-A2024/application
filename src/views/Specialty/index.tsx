import Specialty from '@src/components/organisms/Specialty'
import { CardEvolution } from '@src/types'
import React from 'react'

export default function SpecialtyView({
  context
}: {
  context: {
    isPatient: boolean
    isDoctor: boolean
    specialty: {
      id: string
      name: string
    }
    data: {
      evolutions?: CardEvolution[]
      orders?: CardEvolution[]
      tests?: CardEvolution[]
    }
    currentTab: number
  }
}) {
  return <Specialty context={context} />
}
