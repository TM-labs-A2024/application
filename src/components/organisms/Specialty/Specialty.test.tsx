import { specialties, specialtyData } from '@constants/index'
import { render, screen } from '@test/utils/index'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'

import '@testing-library/jest-dom'
import Specialty from './Specialty'
import 'next/navigation'
import 'next/router'

jest.mock('next/navigation')
jest.mock('next/router', () => ({
  useRouter: () => ({
    pathname: '#',
    push: jest.fn(),
    query: {
      slug: ['3fa85f64-5717-4562-b3fc-2c963f66afa4']
    }
  })
}))

const patientData = {
  evolutions: specialtyData.evolutions.map(({ id, date, type, author, reason }) => ({
    href: `/evolucion/1/${id}`,
    title: `${type}: ${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    description: `Editado por: ${author}`,
    comment: `Patología: ${reason}`
  })),
  orders: specialtyData.orders.map(({ id, title, date, author }) => ({
    href: `/orden/1/${id}`,
    title,
    description: `${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    comment: `Agregado por: ${author}`
  })),
  tests: specialtyData.tests.map(({ id, title, date, author }) => ({
    href: `/analisis/1/${id}`,
    title,
    description: `${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    comment: `Agregado por: ${author}`
  }))
}

const patientContext = {
  isPatient: true,
  isDoctor: false,
  specialty: specialties[0],
  data: patientData,
  currentTab: 0
}

const nurseContext = {
  isPatient: false,
  isDoctor: false,
  specialty: specialties[0],
  data: patientData,
  currentTab: 0
}

const emptyData = {
  evolutions: [],
  orders: [],
  tests: []
}

const emptyContext = {
  isPatient: true,
  isDoctor: false,
  specialty: specialties[0],
  data: emptyData,
  currentTab: 0
}

const doctorData = {
  evolutions: specialtyData.evolutions.map(({ id, date, type, author, reason }) => ({
    href: `/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa4/1/${id}`,
    title: `${type}: ${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    description: `Editado por: ${author}`,
    comment: `Patología: ${reason}`
  })),
  orders: specialtyData.orders.map(({ id, title, date, author }) => ({
    href: `/orden/3fa85f64-5717-4562-b3fc-2c963f66afa4/1/${id}`,
    title,
    description: `${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    comment: `Agregado por: ${author}`
  })),
  tests: specialtyData.tests.map(({ id, title, date, author }) => ({
    href: `/analisis/3fa85f64-5717-4562-b3fc-2c963f66afa4/1/${id}`,
    title,
    description: `${format(new Date(date), 'dd, MMMM yyyy', {
      locale: es
    })}`,
    comment: `Agregado por: ${author}`
  }))
}

const doctorContext = {
  isPatient: false,
  isDoctor: true,
  specialty: specialties[0],
  data: doctorData,
  currentTab: 0
}

describe('Organisms > Specialties test', () => {
  test('The component shows the evolutions list', () => {
    render(<Specialty context={patientContext} />)

    expect(screen.getByText(patientData.evolutions[0].title)).toBeInTheDocument()
  })

  test('The component shows the orders list if it has at least one item', () => {
    render(<Specialty context={patientContext} />)

    expect(screen.getByText(patientData.orders[0].title)).toBeInTheDocument()
  })

  test('The component shows the tests list if it has at least one item', () => {
    render(<Specialty context={patientContext} />)

    expect(screen.getByText(patientData.tests[0].title)).toBeInTheDocument()
  })

  test('The component shows the empty state if the specialties states is empty', () => {
    render(<Specialty context={emptyContext} />)

    expect(screen.getByTestId('orders-empty-state')).toBeInTheDocument()
  })

  test('The component shows the empty state if the specialties states is empty', () => {
    render(<Specialty context={emptyContext} />)

    expect(screen.getByTestId('tests-empty-state')).toBeInTheDocument()
  })

  test("If the component is rendered in the patient's flow, it doesn't show the create evolution, order and test buttons", () => {
    render(<Specialty context={patientContext} />)

    expect(screen.queryByText('Nueva evolución')).not.toBeInTheDocument()
    expect(screen.queryByText('Nueva orden')).not.toBeInTheDocument()
    expect(screen.queryByText('Nuevo análisis')).not.toBeInTheDocument()
  })

  test("If the component is rendered in the dcotors's flow, it does show the create evolution button", () => {
    render(<Specialty context={doctorContext} />)

    expect(screen.getByText('Nueva evolución')).toBeInTheDocument()
  })

  test("If the component is rendered in the dcotors's flow, it does show the create order button", () => {
    render(<Specialty context={{ ...doctorContext, currentTab: 1 }} />)

    expect(screen.getByText('Nueva orden')).toBeInTheDocument()
  })

  test("If the component is rendered in the dcotors's flow, it does show the create test button", () => {
    render(<Specialty context={{ ...doctorContext, currentTab: 2 }} />)

    expect(screen.getByText('Nuevo análisis')).toBeInTheDocument()
  })

  test("If the component is rendered in the nuerses' flow, it only shows the orders list", () => {
    render(<Specialty context={nurseContext} />)

    expect(screen.queryByText('Evoluciones')).not.toBeInTheDocument()
    expect(screen.queryByText('Análisis')).not.toBeInTheDocument()
    expect(screen.getByText('Ordenes médicas')).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<Specialty context={patientContext} />)

    expect(screen.getByTestId('specialty')).toMatchSnapshot()
  })
})
