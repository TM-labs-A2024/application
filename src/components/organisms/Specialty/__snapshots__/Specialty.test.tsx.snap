// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Specialties test Matches the snapshot 1`] = `
<div
  class="flex h-screen w-screen flex-col p-8 lg:px-96 pt-8"
  data-testid="specialty"
>
  <div
    class="mb-8 flex flex-row items-center justify-between"
  >
    <div
      class="flex flex-row items-center justify-start gap-4"
    >
      <button
        aria-label="back"
        class="chakra-button css-84vx7q"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            fill="currentColor"
          />
        </svg>
      </button>
      <p
        class="chakra-text font-medium css-0"
      >
        Alergología
      </p>
    </div>
    <span
      class="chakra-avatar css-xr1dqv"
    >
      <div
        aria-label="Alergología"
        class="chakra-avatar__initials css-1ebyn6"
        role="img"
      >
        A
      </div>
    </span>
  </div>
  <div
    class="mb-8 w-full"
    data-testid="search-input-component"
  >
    <div
      class="chakra-input__group css-1y0e7gb"
      data-group="true"
    >
      <div
        class="chakra-input__left-element css-gv1iwz"
      >
        <svg
          class="chakra-icon css-13izzit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        class="chakra-input min-h-10 css-qluo93"
        placeholder="Buscar"
      />
    </div>
  </div>
  <div
    class="chakra-tabs h-[80%] css-1xpribl"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-1uw5bnh"
      role="tablist"
    >
      <button
        aria-controls="tabs-:ra:--tabpanel-0"
        aria-disabled="false"
        aria-selected="true"
        class="chakra-tabs__tab css-1035tk2"
        data-index="0"
        id="tabs-:ra:--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Evoluciones
      </button>
      <button
        aria-controls="tabs-:ra:--tabpanel-1"
        aria-disabled="false"
        aria-selected="false"
        class="chakra-tabs__tab css-1af0cf2"
        data-index="1"
        id="tabs-:ra:--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Ordenes médicas
      </button>
      <button
        aria-controls="tabs-:ra:--tabpanel-2"
        aria-disabled="false"
        aria-selected="false"
        class="chakra-tabs__tab css-1035tk2"
        data-index="2"
        id="tabs-:ra:--tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Análisis
      </button>
    </div>
    <div
      class="chakra-tabs__tab-indicator css-13g3kif"
      style="position: absolute; transition-property: left, right, top, bottom, height, width; transition-duration: 0ms; transition-timing-function: cubic-bezier(0, 0, 0.2, 1); left: 0px; width: 0px;"
    />
    <div
      class="chakra-tabs__tab-panels h-full pb-0 css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:ra:--tab-0"
        class="chakra-tabs__tab-panel h-[90%] overflow-scroll css-a5mhaz"
        id="tabs-:ra:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        <ul
          class="h-full css-1xa84ef"
          data-testid="evolutions-list"
          role="list"
        >
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/1"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 1: 17, septiembre 2024
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/2"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 2: 17, septiembre 2024
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/3"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 3: 17, septiembre 2024
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/4"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 4: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/5"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 5: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/6"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 6: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/7"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 7: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/8"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 8: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/evolucion/1/9"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Evolución 9: 17, septiembre 1997
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Editado por: Pedro Pérez
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Patología: Dolor
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
      <div
        aria-labelledby="tabs-:ra:--tab-1"
        class="chakra-tabs__tab-panel h-[90%] overflow-scroll css-a5mhaz"
        hidden=""
        id="tabs-:ra:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        <ul
          class="h-full css-1xa84ef"
          data-testid="evolutions-list"
          role="list"
        >
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/1"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 1
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/2"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 2
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/3"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 3
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/4"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 4
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/5"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 5
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/6"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 6
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/7"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 7
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/8"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 8
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/orden/1/9"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Orden 9
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
      <div
        aria-labelledby="tabs-:ra:--tab-2"
        class="chakra-tabs__tab-panel h-[90%] overflow-scroll css-a5mhaz"
        hidden=""
        id="tabs-:ra:--tabpanel-2"
        role="tabpanel"
        tabindex="0"
      >
        <ul
          class="h-full css-1xa84ef"
          data-testid="evolutions-list"
          role="list"
        >
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/1"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 1
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/2"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 2
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/3"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 3
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/4"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 4
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/5"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 5
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/6"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 6
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/7"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 7
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/8"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 8
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black last:border-transparent"
          >
            <a
              class="chakra-link w-full css-ofs0kf"
              data-testid="evolution-card"
              href="/analisis/1/9"
            >
              <div
                class="mb-1 flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Análisis 9
                  </b>
                  <div
                    class="flex flex-col items-start gap-1"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      17, septiembre 1997
                    </p>
                    <p
                      class="chakra-text css-0"
                    >
                      Agregado por: Pedro Pérez
                    </p>
                  </div>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`
